inference/models/inference.py

```diff
@@ -0,0 +1,292 @@
+    description = models.TextField()
+    classification_type = models.IntegerField(
+        choices=ClassificationType.choices, help_text="Type of classification this model performs"
+    )
+    is_active = models.BooleanField(
+        default=True, help_text="Whether this is the current active version for its classification type"
+    )
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=["classification_type", "is_active"],
+                condition=models.Q(is_active=True),
+                name="unique_active_version",
+            )
+        ]
+
+    def __str__(self):
+        return f"{self.get_classification_type_display()} - {self.api_identifier}"
+
+    @classmethod
+    def get_active_version(cls, classification_type: int) -> "ModelVersion":
+        """Get the current active model version for a classification type."""
+        return cls.objects.get(classification_type=classification_type, is_active=True)
+
+    def set_as_active(self):
+        """Set this version as the active one for its classification type."""
+        # Deactivate other versions of this classification type
+        ModelVersion.objects.filter(classification_type=self.classification_type, is_active=True).exclude(
+            id=self.id
+        ).update(is_active=False)
+
+        # Set this one as active
+        self.is_active = True
+        self.save()
+
+
+class InferenceJob(models.Model):
+    """
+    Tracks an inference job for a collection of URLs.
+    One InferenceJob can have multiple ExternalJobs (one per batch).
+    """
+
+    collection = models.ForeignKey(
+        "sde_collections.Collection", on_delete=models.CASCADE, related_name="inference_jobs"
+    )
+    model_version = models.ForeignKey(
+        ModelVersion,
+        on_delete=models.PROTECT,  # Prevent deletion of ModelVersions that have associated jobs
+        related_name="inference_jobs",
+    )
+
+    created_at = models.DateTimeField(default=timezone.now)
+    updated_at = models.DateTimeField(auto_now=True)
+    completed_at = models.DateTimeField(null=True, blank=True)
+
+    status = models.IntegerField(choices=InferenceJobStatus.choices, default=InferenceJobStatus.QUEUED)
+    error_message = models.TextField(blank=True)
+
+    class Meta:
+        ordering = ["-created_at"]
+        constraints = [
+            models.UniqueConstraint(
+                fields=["collection", "model_version"],
+                condition=models.Q(status__in=[InferenceJobStatus.QUEUED, InferenceJobStatus.PENDING]),
+                name="unique_active_job",
+            )
+        ]
+
+    def __str__(self):
+        return f"Job {self.id} - {self.collection} - {self.model_version}"
+
+    @property
+    def is_ongoing(self) -> bool:
+        """Check if job still has active external jobs"""
+        return self.external_jobs.filter(status__in=[ExternalJobStatus.QUEUED, ExternalJobStatus.PENDING]).exists()
```

Comment:
This might be confusing if you see Inference.status=Queued (and therefor no jobs made) but inference.is_ongoing=False?

Or is a queued one technically not running? Maybe we can change from ongoing to running?